server/server.go

```diff
@@ -0,0 +1,60 @@
+package server
+
+import (
+	"encoding/json"
+	"net/http"
+	"os"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+//go:generate mockery --name=StorageAdapter --inpackage --filename=storage_adapter_mock.go
+type StorageAdapter interface {
+	StoreHostTelemetry(*HostTelemetry) error
+}
+
+type HostTelemetry struct {
+	AgentID       string    `json:"agent_id"`
+	SLESVersion   string    `json:"sles_version"`
+	CPUCount      int       `json:"payload"`
+	SocketCount   int       `json:"socket_count"`
+	TotalMemoryMB int       `json:"total_memory_mb"`
+	CloudProvider string    `json:"cloud_provider"`
+	Time          time.Time `json:"time"`
+}
+
+func hostTelemetryHandler(adapters ...StorageAdapter) func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Maybe using `gin` would be better, to keep consistency. Let's leave it for a future discussion hehe

Reply:
Yep, I just used the standard library since it was very minimal. I'll keep this on the radar.